src/gsi/balmod.f90

```diff
@@ -483,9 +528,63 @@ subroutine prebal_reg(cwcoveqqcov)
              end do
           end do
        end do
+
+       if( if_cs_staticB )then
+          call gsi_bundlegetpointer(gsi_metguess_bundle(1),'mask',ges_mask,istatus)
+          if(extra_var2num>0 )then
+             do ivar = 1, extra_var2num
+                do k=1,nsig
+                   do j=1,lon2
+                      do i=1,lat2
+                         l=int(rllat1(i,j))
+                         l2=min0(l+1,llmax)
+                         dl2=rllat1(i,j)-float(l)
+                         dl1=one-dl2
+                         if( ges_mask(i,j,k) < -0.5_r_kind   )then
+                            evik2_real(ivar,i,j,k)=dl1*evi2(ivar,l,2,k)+dl2*evi2(ivar,l2,2,k)
+                         else if ( ges_mask(i,j,k) > 0.5_r_kind  )then
+                            evik2_real(ivar,i,j,k)=dl1*evi2(ivar,l,3,k)+dl2*evi2(ivar,l2,3,k)
+                         else
+                            evik2_real(ivar,i,j,k)=dl1*evi2(ivar,l,1,k)+dl2*evi2(ivar,l2,1,k)
+                         end if
+                      end do
+                   end do
+                end do
+             end do
+          end if
+
+
+          do ivar = 1, extra_var3num
+             do k=1,nsig
+                do m=1,nsig
+                   do j=1,lon2
+                      do i=1,lat2
+                         l=int(rllat1(i,j))
+                         l2=min0(l+1,llmax)
+                         dl2=rllat1(i,j)-float(l)
+                         dl1=one-dl2
+                         if( ges_mask(i,j,k) < -0.5_r_kind  )then
+                            evik3_real(ivar,i,j,m,k)=dl1*evi3(ivar,l,2,m,k)+dl2*evi3(ivar,l2,2,m,k)
+                         else if ( ges_mask(i,j,k) > 0.5_r_kind  )then
+                            evik3_real(ivar,i,j,m,k)=dl1*evi3(ivar,l,3,m,k)+dl2*evi3(ivar,l2,3,m,k)
+                         else
+                            evik3_real(ivar,i,j,m,k)=dl1*evi3(ivar,l,1,m,k)+dl2*evi3(ivar,l2,1,m,k)
+                         end if
```

Comment:
> If you decide to bin evik2_real(:,i,j,k) by ges_mask(i,j,k), I think evik3_real(:,i,j,m,k) should be binned both by ges_mask(i,j,k) and by ges_mask(i,j,m). Why did you use only ges_mask(i,j,k)?

ges_mask is defined based on the composite reflectivity, so it is fine to use ges_mask(i,j,k).

Reply:
Does it mean that ges_mask is one of the 2D variable? The GSI-mask code doesn't look so, but if it is exactly the 2D variable, ges_mask should not depend on k and it should be ges_mask( : , : ).

Reply:
It is correct, but the current GSI code doesn't work properly unless all levels have the same values in ges_mask. I think it causes confusion even if the python script is used instead of the NCL code. If it is impossible to make ges_mask the 2D variable, I suggest binning as follows in GSI. Similar modification might be required also for ges_mask1.
```
if( maxval(ges_mask(i,j,:)) > 90.0_r_kind )then
   (use 3rd bin)
else if ( minval(ges_mask(i,j,:)) < -90.0_r_kind )then
   (use 2nd bin)
else
   (use 1st bin)
endif
```